distributed/tensor_parallelism/example.py

```diff
@@ -74,30 +107,48 @@ def forward(self, x):
         return self.net2(self.relu(self.net1(x)))
 
 
-def demo_tp(rank, args):
+def generate_mesh(world_size, is_2d):
+    mesh = torch.arange(0, world_size)
+    if not is_2d:
+        return mesh
+    else:
+        assert world_size % 2 == 0, "Need to have 2N GPU to enable 2D demo."
+        # TP_size = 2.
+        return mesh.view(-1, 2)
+
+
+def demo(rank, args):
     """
     Main body of the demo of a basic version of tensor parallel by using
     PyTorch native APIs.
     """
     print(f"Running basic Megatron style TP example on rank {rank}.")
     setup(rank, args.world_size)
+
     # create a sharding plan based on the given world_size.
-    device_mesh = DeviceMesh(
-        "cuda",
-        torch.arange(args.world_size),
-    )
+    device_mesh = DeviceMesh("cuda", generate_mesh(args.world_size, args.run_2d))
```

Comment:
Maybe we should have a separate 2-D example too? I think for examples, we should have a relatively standalone simple script without too much helper functions, so that user can easily understand how the API works